sympy/matrices/dense.py

```diff
@@ -164,12 +164,151 @@ def matrix2numpy(m, dtype=object):  # pragma: no cover
+            \end{bmatrix}
+
+    Where $c = \cos(\theta)$ and $s = \sin(\theta)$ appear at the intersections
+    ``i``\th and ``j``\th rows and columns.
+
+    For fixed ``i > j``\, the non-zero elements of a Givens matrix are given by:
+
+    .. math::
+        \begin{align}
+            g_{kk} &= 1 \qquad \text{for} \ k \ne i,\,j \\
+            g_{kk} &= c \qquad \text{for} \ k = i,\,j \\
+            g_{ji} &= -g_{ij} = -s
+        \end{align}
+
+    Parameters
+    ==========
+
+    i : int between 1 and dim
+        Represents first axis
+    j : int between 1 and dim
+        Represents second axis
+    dim : int bigger than 1
+        Number of dimentions. Defaults to 3.
+
+    Examples
+    ========
+
+    >>> from sympy import pi, rot_givens
+
+    A counterclockwise rotation of pi/3 (60 degrees) around
+    the third axis (z-axis):
+
+    >>> rot_givens(1, 2, pi/3)
+    Matrix([
+    [      1/2, -sqrt(3)/2, 0],
+    [sqrt(3)/2,        1/2, 0],
+    [        0,          0, 1]])
+
+    If we rotate by pi/2 (90 degrees):
+
+    >>> rot_givens(1, 2, pi/2)
+    Matrix([
+    [0, -1, 0],
+    [1,  0, 0],
+    [0,  0, 1]])
+
+    This can be generalized to any number
+    of dimentions:
+
+    >>> rot_givens(1, 2, pi/2, dim=4)
+    Matrix([
+    [0, -1, 0, 0],
+    [1,  0, 0, 0],
+    [0,  0, 1, 0],
+    [0,  0, 0, 1]])
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Givens_rotation
+
+    See Also
+    ========
+
+    rot_axis1: Returns a rotation matrix for a rotation of theta (in radians)
+        about the 1-axis (clockwise around the x axis)
+    rot_axis2: Returns a rotation matrix for a rotation of theta (in radians)
+        about the 2-axis (clockwise around the y axis)
+    rot_axis3: Returns a rotation matrix for a rotation of theta (in radians)
+        about the 3-axis (clockwise around the z axis)
+    rot_ccw_axis1: Returns a rotation matrix for a rotation of theta (in radians)
+        about the 1-axis (counterclockwise around the x axis)
+    rot_ccw_axis2: Returns a rotation matrix for a rotation of theta (in radians)
+        about the 2-axis (counterclockwise around the y axis)
+    rot_ccw_axis3: Returns a rotation matrix for a rotation of theta (in radians)
+        about the 3-axis (counterclockwise around the z axis)
+    """
+    if not isinstance(dim, int) or dim < 2:
+        raise ValueError('dim must be an integer biggen than one, '
+                         'got {}.'.format(dim))
+
+    for ij in [i, j]:
+        if not isinstance(ij, int) or ij < 1 or ij > dim:
+            raise ValueError('i and j must be between 1 and '
+                             '{}, got i={} and j={}.'.format(dim, i, j))
+
+    theta = sympify(theta)
+    c = cos(theta)
+    s = sin(theta)
+    M = eye(dim)
+    if i < j:
+        i, j = j, i
+        theta = -theta
+
+    M[i-1, i-1] = c
+    M[j-1, j-1] = c
+    M[i-1, j-1] = s
+    M[j-1, i-1] = -s
```

Comment:
I think that 0 to n-1 indexing is more preferable because of having less complicated indexing

Reply:
Alright, I can change that tomorrow. 

Isn't it better to keep it consistent with axis1, axis2 and axis3 though?

Reply:
No, those methods are pretty much nominal, they have only a finite numbers of names, so they can be left as is.
However a program that generates infinite numbers of notations is more general and can be difficult to change in the future.
It is better to be consistent with python indexing, or sympy matrix indexing itself, or there would be other counter arguments.

Reply:
Fair, just changed it!